src/main/java/info/skyblond/nekohit/neo/contract/WCAContract.java

```diff
@@ -177,59 +182,51 @@ public static String createWCA(
             int thresholdIndex, int coolDownInterval,
             boolean bePublic, String identifier
     ) throws Exception {
-        require(Runtime.checkWitness(owner) || owner == Runtime.getCallingScriptHash(), "Invalid sender signature. The owner of the wca needs to be the signing account.");
+        require(Runtime.checkWitness(owner) || owner == Runtime.getCallingScriptHash(), Messages.INVALID_SIGNATURE);
         // identifier should be unique
-        require(wcaBasicInfoMap.get(identifier) == null, "Duplicate identifier.");
+        require(wcaBasicInfoMap.get(identifier) == null, Messages.DUPLICATED_ID);
         // check milestone
-        require(milestoneTitles.length == milestoneDescriptions.length, "Cannot decide milestones count.");
-        require(milestoneDescriptions.length == endTimestamps.length, "Cannot decide milestones count.");
+        require(milestoneTitles.length == milestoneDescriptions.length, Messages.INVALID_MILESTONES_COUNT);
+        require(milestoneDescriptions.length == endTimestamps.length, Messages.INVALID_MILESTONES_COUNT);
 
         // convert to object on the fly
         List<WCAMilestone> milestones = new List<>();
         int lastTimestamp = 0;
         for (int i = 0; i < endTimestamps.length; i++) {
-            require(lastTimestamp < endTimestamps[i], "The end timestamp should increase.");
-            require(endTimestamps[i] > Runtime.getTime(), "The end timestamp is already expired.");
+            require(lastTimestamp < endTimestamps[i], Messages.INVALID_TIMESTAMP);
+            require(endTimestamps[i] > Runtime.getTime(), Messages.EXPIRED_TIMESTAMP);
             lastTimestamp = endTimestamps[i];
             milestones.add(new WCAMilestone(milestoneTitles[i], milestoneDescriptions[i], endTimestamps[i]));
         }
 
         // create wca info obj
-        WCABasicInfo info = new WCABasicInfo(
+        WCABasicInfo basicInfo = new WCABasicInfo(
                 owner, wcaDescription, stakePer100Token, maxTokenSoldCount,
                 milestones.size(), thresholdIndex, coolDownInterval, bePublic
         );
-
-        ByteString basicData = StdLib.serialize(info);
-        ByteString milestoneData = StdLib.serialize(milestones);
-        ByteString buyerData = StdLib.serialize(new WCABuyerInfo(maxTokenSoldCount));
-
         // store
-        wcaBasicInfoMap.put(identifier, basicData);
-        wcaMilestonesMap.put(identifier, milestoneData);
-        wcaBuyerInfoMap.put(identifier, buyerData);
-
+        updateWCABasicInfo(identifier, basicInfo);
+        updateWCABuyerInfo(identifier, new WCABuyerInfo(maxTokenSoldCount));
```

Comment:
maybe we should move the creation of the BuyerInfo to the first-time payment of the creator. The creator can decide to cancel the WCA right after creating it. This will result in a higher GAS cost than when we wait with the creation of the buyer info until he paid it.

Reply:
#24